ios/Runner/AppDelegate.swift

```diff
@@ -15,18 +15,34 @@ import FirebaseCore
         
         FirebaseApp.configure()
         
+        var modifiedLaunchOptions = launchOptions
+        
+        if let remoteNotification = launchOptions?[UIApplication.LaunchOptionsKey.remoteNotification] as? [String: Any],
+           let cioMap = remoteNotification["CIO"] as? [String: Any],
+           let pushMap = cioMap["push"] as? [String: Any],
+           let deepLinkURL = pushMap["link"] as? String {
+            if let url = launchOptions?[UIApplication.LaunchOptionsKey.url] as? String {
+                UIPasteboard.general.string = url
+            }
+            if let url = URL(string: deepLinkURL) {
+                let path = url.path
+                let flutterViewController = window?.rootViewController as! FlutterViewController
+                // Sets path for Flutter router
+                flutterViewController.setInitialRoute(path)
```

Comment:
[`setInitialRoute`](https://api.flutter.dev/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)setInitialRoute:) is deprecated. I tried different solutions e.g. [`pushRoute`](https://docs.flutter.dev/add-to-app/ios/add-flutter-screen?tab=no-engine-vc-uikit-swift-tab#route), and creating new `FlutterEngine`, find and update current `FlutterEngine` path, insert new `FlutterViewController`, but nothing worked for me. So I have used the deprecated api for now.

Reply:
That's not needed when app is running. GoRouter automatically handles it and routes to the correct path. So far, notification links works fine in foreground, background and killed state.